cpp/src/arrow/array/validate.cc

```diff
@@ -166,6 +166,86 @@ struct ValidateArrayImpl {
     return Status::OK();
   }
 
+  Status Visit(const Date64Type& type) {
+    // check that data is divisible by 8.64e7 (= 1000ms * 60s * 60mins * 24hrs)
+    RETURN_NOT_OK(ValidateFixedWidthBuffers());
+    
+    using c_type = typename Date64Type::c_type;
+    if (full_validation) {
+      c_type fullDay = 1000 * 60 * 60 * 24;
```

Comment:
`constexpr c_type kFullDayMillis`? (Constants in the [Google style guide](https://arrow.apache.org/docs/developers/cpp/development.html#code-style-linting-and-ci) use [`kCamelCase`](https://google.github.io/styleguide/cppguide.html#Constant_Names).)